### DISCUSSION TITLE
Query regarding .env file to configure NC_DB

### DISCUSSION THREAD

**Discussion by MGW-SumathyV** [nocodb/nocodb#2665] [Q&A]
Hi,

I created development setup for running nocodb in my local machine.
I created a .env file under \nocodb\packages\nocodb and added an entry for NC_DB pointing to my local mysql DB. 
Then I build  using "npm run watch:build". But I am getting below error.

> nocodb@0.92.3 build:obfuscate D:\OMSHREESAIRAM\LotusSimplia\nocodb\packages\nocodb
> EE=true webpack  --config webpack.config.js

'EE' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nocodb@0.92.3 build:obfuscate: `EE=true webpack  --config webpack.config.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nocodb@0.92.3 build:obfuscate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

> nocodb@0.92.3 build:main D:\OMSHREESAIRAM\LotusSimplia\nocodb\packages\nocodb
> tsc -p tsconfig.json

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sumathy\AppData\Roaming\npm-cache\_logs\2022-07-13T06_50_21_979Z-debug.log

> nocodb@0.92.3 build:module D:\OMSHREESAIRAM\LotusSimplia\nocodb\packages\nocodb
> tsc -p tsconfig.module.json

ERROR: "build:obfuscate" exited with 1.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nocodb@0.92.3 build: `run-s clean && run-p build:*`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nocodb@0.92.3 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sumathy\AppData\Roaming\npm-cache\_logs\2022-07-13T06_50_22_197Z-debug.log
[nodemon] app crashed - waiting for file changes before starting...
  

Could you please help in resolving the issue?

Thanks and Regards,
Sumathy Venugopal

**Comment by wingkwong** [comment]
If you are using Windows, you may need to add `cross-env` before those commands.

e.g.

from

```
"build:obfuscate": "EE=true webpack  --config webpack.config.js",
```

to 

```
"build:obfuscate": "cross-env EE=true webpack  --config webpack.config.js",
```

**Comment by MGW-SumathyV** [comment]
Previous Error is resolved .But I am getting below error

WARNING in ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
Module not found: Error: Can't resolve './templates/mssql.template.js' in 'D:\nocodb\packages\nocodb\src\lib\db\sql-migrator\lib'
 @ ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
 @ ./src/lib/Noco.ts
 @ ./src/lib/index.ts

WARNING in ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
Module not found: Error: Can't resolve './templates/mysql.template.js' in 'D:\nocodb\packages\nocodb\src\lib\db\sql-migrator\lib'
 @ ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
 @ ./src/lib/Noco.ts
 @ ./src/lib/index.ts

WARNING in ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
Module not found: Error: Can't resolve './templates/oracle.template.js' in 'D:\nocodb\packages\nocodb\src\lib\db\sql-migrator\lib'
 @ ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
 @ ./src/lib/Noco.ts
 @ ./src/lib/index.ts

WARNING in ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
Module not found: Error: Can't resolve './templates/pg.template.js' in 'D:\nocodb\packages\nocodb\src\lib\db\sql-migrator\lib'
 @ ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
 @ ./src/lib/Noco.ts
 @ ./src/lib/index.ts

WARNING in ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
Module not found: Error: Can't resolve './templates/sqlite.template.js' in 'D:\nocodb\packages\nocodb\src\lib\db\sql-migrator\lib'
 @ ./src/lib/db/sql-migrator/lib/KnexMigrator.ts
 @ ./src/lib/Noco.ts
 @ ./src/lib/index.ts

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/

> nocodb@0.92.3 postbuild D:\nocodb\packages\nocodb
> npm run copy-files


> nocodb@0.92.3 copy-files D:\nocodb\packages\nocodb
> copyfiles -u 1 "src/**/*.ejs" build/main && copyfiles -u 1 "src/**/*.ejs" build/module && copyfiles -u 1 "src/**/*.ejs" docker

[nodemon] clean exit - waiting for changes before restart

**Reply by wingkwong** [reply]
Can you show the steps?

**Comment by MGW-SumathyV** [comment]
I referred the development setup given in the README doc of nocodb.

**Cloning the Project**

```shell
git clone https://github.com/nocodb/nocodb
cd nocodb
```

**## Build SDK**

```shell
cd packages/nocodb-sdk
npm install
npm run build
```

**## Running Backend locally**

```shell
cd packages/nocodb
npm install

I created a .env file with NC_DB entry 
npm run watch:build

**Reply by wingkwong** [reply]
Please also share your npm & node version.

**Reply by MGW-SumathyV** [reply]
node 14.17.6
npm 6.14.15

**Reply by wingkwong** [reply]
The setup and steps look fine. Can you also share your `.env`?

**Reply by MGW-SumathyV** [reply]
NC_DB = "mysql://localhost:3306?u=ru&p=su&d=sumathyschema"

This is the only entry I added in .env

**Reply by wingkwong** [reply]
Try the following instead.

```
NC_DB=mysql://localhost:3306?u=ru&p=su&d=sumathyschema
```

**Reply by MGW-SumathyV** [reply]
I am getting the same error

**Reply by MGW-SumathyV** [reply]
I got the new version of nocodb and tried with .env file .But I am getting below error

ERROR in ./src/lib/db/sql-client/lib/KnexClient.ts
Module not found: Error: Can't resolve 'faker' in 'D:\OMSHREESAIRAM\LotusSimplia\nocodb\packages\nocodb\src\lib\db\sql-client\lib'
 @ ./src/lib/db/sql-client/lib/KnexClient.ts 856:22-38
 @ ./src/lib/db/sql-client/lib/oracle/OracleClient.ts
 @ ./src/lib/db/sql-client/lib/SqlClientFactory.ts
 @ ./src/lib/utils/NcConfigFactory.ts
 @ ./src/lib/index.ts
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! nocodb@0.92.3 build:obfuscate: `cross-env EE=true webpack  --config webpack.config.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the nocodb@0.92.3 build:obfuscate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sumathy\AppData\Roaming\npm-cache\_logs\2022-07-14T06_47_05_467Z-debug.log
ERROR: "build:obfuscate" exited with 2.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nocodb@0.92.3 build: `run-s clean && run-p build:*`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the nocodb@0.92.3 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sumathy\AppData\Roaming\npm-cache\_logs\2022-07-14T06_47_06_137Z-debug.log
[nodemon] app crashed - waiting for file changes before starting...

**Reply by wingkwong** [reply]
Try removing node_modules then run npm install again.

**Reply by MGW-SumathyV** [reply]
I tried but getting same error
